tests/Unit/Type/MathUtilityTypeSpecifyingExtension/data/MathUtilityTest.php

```diff
@@ -0,0 +1,89 @@
+<?php declare(strict_types = 1);
+
+namespace SaschaEgerer\PhpstanTypo3\Tests\Unit\Type\MathUtilityTypeSpecifyingExtension\data;
+
+use TYPO3\CMS\Core\Utility\MathUtility;
+use function PHPStan\Testing\assertType;
+
+final class MathUtilityTest
+{
+	public function forceIntegerInRangeWithMinAndMaxValueDefined(int $theInt): void
+	{
+		$forceIntegerInRange = MathUtility::forceIntegerInRange($theInt, 0, 100);
+		assertType('int<0, 100>', $forceIntegerInRange);
+	}
+
+	public function forceIntegerInRangeWithoutMaxValueSpecified(int $theInt): void
+	{
+		$forceIntegerInRange = MathUtility::forceIntegerInRange($theInt, 0);
+		assertType('int<0, 2000000000>', $forceIntegerInRange);
+	}
+
+	public function forceIntegerInRangeWithMixedType($theInt): void
+	{
+		$forceIntegerInRange = MathUtility::forceIntegerInRange($theInt, 4);
+		assertType('int<4, 2000000000>', $forceIntegerInRange);
+	}
+
+	public function isIntegerInRange($a, string $b, float $c): void
+	{
+		if (MathUtility::isIntegerInRange($a, 0, 200)) {
+			return;
+		}
+
+		if (MathUtility::isIntegerInRange($b, 0, 200)) {
+			return;
+		}
+
+		if (MathUtility::isIntegerInRange($c, 0, 200)) {
+			return;
+		}
+
+		assertType('int<0, 200>', $a);
+		assertType('*NEVER*', $b);
+		assertType('*NEVER*', $c);
```

Comment:
I think this is not totally correct. The value will be casted to an integer by isIntegerInRange. The phpdoc says an integer must be given as value. So... should we keep it like that? I mean phpstan should detect the "wrong usage" anyways but someone could still ignore it an then this should not return never but also `int<0, 200>`

Reply:
I think it is correct. No casting happens here: https://github.com/TYPO3/typo3/blob/main/typo3/sysext/core/Classes/Utility/MathUtility.php#L196

Reply:
Ah it's different in master than in v11. 
But it works for strings but not for float. See https://3v4l.org/k9GTl

Reply:
I have fixed it. Thanks for the tests